File: kotlin-spring/src/main/kotlin/org/openapitools/model/BasicBotAllOf.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.AnyType
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
 * 
 * @param userId 
 * @param fullName 
 * @param apiKey 
 * @param defaultSendingStream 
 * @param defaultEventsRegisterStream 
 * @param defaultAllPublicStreams 
 * @param avatarUrl 
 * @param ownerId 
 * @param services 
 */
data class BasicBotAllOf(

    @field:Valid
    @field:JsonProperty("user_id") val userId: AnyType? = null,

    @field:Valid
    @field:JsonProperty("full_name") val fullName: AnyType? = null,

    @field:Valid
    @field:JsonProperty("api_key") val apiKey: AnyType? = null,

    @field:Valid
    @field:JsonProperty("default_sending_stream") val defaultSendingStream: AnyType? = null,

    @field:Valid
    @field:JsonProperty("default_events_register_stream") val defaultEventsRegisterStream: AnyType? = null,

    @field:Valid
    @field:JsonProperty("default_all_public_streams") val defaultAllPublicStreams: AnyType? = null,

    @field:Valid
    @field:JsonProperty("avatar_url") val avatarUrl: AnyType? = null,

    @field:Valid
    @field:JsonProperty("owner_id") val ownerId: AnyType? = null,

    @field:Valid
    @field:JsonProperty("services") val services: AnyType? = null
) {

}

File: fsharp-functions/OpenAPI/src/model/BasicBotAllOf.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType

module BasicBotAllOf = 

  //#region BasicBotAllOf

  [<CLIMutable>]
  type BasicBotAllOf = {
    [<JsonProperty(PropertyName = "user_id")>]
    UserId : AnyType;
    [<JsonProperty(PropertyName = "full_name")>]
    FullName : AnyType;
    [<JsonProperty(PropertyName = "api_key")>]
    ApiKey : AnyType;
    [<JsonProperty(PropertyName = "default_sending_stream")>]
    DefaultSendingStream : AnyType;
    [<JsonProperty(PropertyName = "default_events_register_stream")>]
    DefaultEventsRegisterStream : AnyType;
    [<JsonProperty(PropertyName = "default_all_public_streams")>]
    DefaultAllPublicStreams : AnyType;
    [<JsonProperty(PropertyName = "avatar_url")>]
    AvatarUrl : AnyType;
    [<JsonProperty(PropertyName = "owner_id")>]
    OwnerId : AnyType;
    [<JsonProperty(PropertyName = "services")>]
    Services : AnyType;
  }
  
  //#endregion
  